rmgpy/reaction.py

```diff
@@ -787,6 +788,28 @@ def reverse_arrhenius_rate(self, k_forward, reverse_units, Tmin=None, Tmax=None)
         kr.fit_to_data(Tlist, klist, reverse_units, kf.T0.value_si)
         return kr
 
+    def reverse_surface_arrhenius_rate(self, k_forward, reverse_units, Tmin=None, Tmax=None):
+        """
+        Reverses the given k_forward, which must be a SurfaceArrhenius type.
+        You must supply the correct units for the reverse rate.
+        The equilibrium constant is evaluated from the current reaction instance (self).
+        """
+        cython.declare(kf=SurfaceArrhenius, kr=SurfaceArrhenius)
+        cython.declare(Tlist=np.ndarray, klist=np.ndarray, i=cython.int)
+        kf = k_forward
+        assert type(kf) is SurfaceArrhenius  # Only reverses SurfaceArrhenius rates
```

Comment:
Instead of assert, do you mind do a `isinstance` check and raising an exception? For example:

```
if not isinstance(kf, SurfaceArrhenius):
    raise TypeError(f'Expected a SurfaceArrhenius object for k_forward but received {kf}')
```